src/bin/ion/commands/generate/generator.rs

```diff
@@ -277,19 +278,55 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         schema_system: &mut SchemaSystem,
     ) -> CodeGenResult<()> {
         for authority in authorities {
-            // Sort the directory paths to ensure nested type names are always ordered based
-            // on directory path. (nested type name uses a counter in its name to represent that type)
-            let mut paths = fs::read_dir(authority)?.collect::<Result<Vec<_>, _>>()?;
-            paths.sort_by_key(|dir| dir.path());
-            for schema_file in paths {
-                let schema_file_path = schema_file.path();
-                let schema_id = schema_file_path.file_name().unwrap().to_str().unwrap();
-
-                let schema = schema_system.load_isl_schema(schema_id).unwrap();
+            self.generate_code_for_directory(authority, None, schema_system)?;
```

Comment:
Can you explain the purpose of this particular change? Is it just a refactoring, or is there a technical reason why it was necessary?

Reply:
It allows traversing through sub directories of the authority. Just created separate method that I cna reuse for each directory getting traversed.